guides/v2.1/extension-dev-guide/versioning/codebase-changes.md

```diff
@@ -108,14 +109,19 @@ Use this table to understand what changes Magento can make and which version num
 | | Changed event property | MAJOR|
 | | Removed event property | MAJOR|
 | | Changed event ordering | MAJOR|
-| **Virtual Type**| Virtual type removed | MAJOR|
+| **Virtual Type (marked with `@api`)**| Virtual type removed | MAJOR|
 | | Virtual type added | MINOR|
-| **URL Paths** | Path removed | MAJOR|
-| | Removed/renamed a request parameter| MAJOR|
-| | New required request parameter | MAJOR|
-| | New optional request parameter | MINOR|
+| **URL Paths (Store-Front and Admin)** | Path removed | MAJOR|
+| | Removed/renamed a request parameter (GET)| MAJOR|
+| | New required request parameter (GET) | MAJOR|
+| | New optional request parameter (GET) | MINOR|
```

Comment:
@buskamuza  considering a new optional request parameter (GET/POST) as MINOR change technically blocking delivery a significant number of bug fixes in the patch release. I agree that the new parameter adds new or change somehow current behavior,  but the behavior can be changed without adding this parameter as well. I believe we should apply this limitation only for URL paths that intended for access by 3-rd party extensions or there is a high probability of such case

Reply:
Will discuss with the community. By default, I propose to consider such changes MINOR/MAJOR for now.

Reply:
@viktym Do we have a way of marking which URL paths are intended for 3rd-party access?

Reply:
We don't currently. So all of therm are public API.

Reply:
Could we define @api URL paths first and only then apply limitations regarding optional request parameters? It will help to avoid blocking of delivery bugfixes in patch releases

Reply:
> We don't currently. So all of them are public API.

In that case, the new URL path should be treated as a MINOR also since some extensions can start using it.
So we are not allowed to add in a patch a new URL-path or optional request parameter. It's a strong limitation.
I believe it should be applied only to @api URL-path.